### PR TITLE
[needs QA] migrate to Aiogram3

### PR DESCRIPTION
**Pull request by bomzheg**
we need help: test all commands and other interactive

### CODE REVIEW

File: app/handlers/errors.py [hunk 1]
```diff
@@ -1,39 +1,44 @@
...
+    except TelegramBadRequest as e:
         if "rights" in e.args[0] and "send" in e.args[0]:
-            if update.message and update.message.chat:
-                logger.info("bot are muted in chat {chat}", chat=update.message.chat.id)
+            if error.update.message and error.update.message.chat:
+                logger.info("bot are muted in chat {chat}", chat=error.update.message.chat.id)
             else:
-                logger.info("bot can't send message (no rights) in update {update}", update=update)
+                logger.info("bot can't send message (no rights) in update {update}", update=error.update)
             return True
```
Comment (Eclips4) [inline app/handlers/errors.py:27]: In aiogram 3.x we should remove this.
This was neccessary in 2.x.

File: app/middlewares/db_middleware.py [hunk 2]
```diff
@@ -18,17 +16,25 @@
...
+        return await handler(event, data)
 
     async def setup_chat(self, data: dict, user: types.User, chat: Optional[types.Chat] = None):
         try:
-            async with self.lock_factory.get_lock(f"{user.id}"):
+            async with self.lock_factory.get_lock(str(user.id)):
                 user = await User.get_or_create_from_tg_user(user)
             if chat and chat.type != 'private':
-                async with self.lock_factory.get_lock(f"{chat.id}"):
+                async with self.lock_factory.get_lock(str(chat.id)):
```
Comment (Eclips4) [inline app/middlewares/db_middleware.py:37]: Does it really need cast to `str`?

File: app/models/db/user.py [hunk 3]
```diff
@@ -66,7 +67,7 @@ async def update_user_data(self, user_tg):
             await self.save()
 
     @classmethod
-    async def get_or_create_from_tg_user(cls, user_tg: types.User):
+    async def get_or_create_from_tg_user(cls, user_tg: types.User | dto.TargetUser):
```
Comment (Eclips4) [inline app/models/db/user.py:70]: Such typehint will not allow start bot on old python versions, such as 3.9-3.7

File: app/services/find_target_user.py [hunk 4]
```diff
@@ -15,15 +16,15 @@
 logger = Logger(__name__)
 
 
-def get_target_user(message: types.Message, can_be_same=False, can_be_bot=False) -> typing.Optional[types.user.User]:
+def get_target_user(message: types.Message, can_be_same=False, can_be_bot=False) -> typing.Optional[dto.TargetUser]:
```
Comment (Eclips4) [inline app/services/find_target_user.py:19]: We need to choose. Optional as `x | None` or `Optional[x]`. Currently it seems inconsistent

File: app/services/find_target_user.py [hunk 5]
```diff
@@ -40,7 +41,7 @@ def get_target_user(message: types.Message, can_be_same=False, can_be_bot=False)
     return None
 
 
-def has_target_user(target_user: types.User, author_user: types.User, can_be_same, can_be_bot):
+def has_target_user(target_user: dto.TargetUser, author_user: dto.TargetUser, can_be_same, can_be_bot):
```
Comment (Eclips4) [inline app/services/find_target_user.py:44]:
```suggestion
def has_target_user(target_user: dto.TargetUser, author_user: dto.TargetUser, can_be_same, can_be_bot) -> bool:
```

File: app/services/moderation.py [hunk 6]
```diff
@@ -125,17 +125,17 @@ def get_duration(text: str):
 
 async def user_has_now_ro(user: User, chat: Chat, bot: Bot):
```
Comment (Eclips4) [inline app/services/moderation.py:126]:
```suggestion
async def user_has_now_ro(user: User, chat: Chat, bot: Bot) -> bool:
```

File: app/services/moderation.py [hunk 7]
```diff
@@ -164,7 +164,7 @@ async def auto_restrict(target: User, chat: Chat, bot: Bot, using_db=None) -> ty
 async def get_count_auto_restrict(target: User, chat: Chat, bot_user: User = None, bot: Bot = None):
```
Comment (Eclips4) [inline app/services/moderation.py:164]: This is a incorrect typehints.
```suggestion
async def get_count_auto_restrict(target: User, chat: Chat, bot_user: typing.Optional[User] = None, bot: typing.Optional[Bot] = None) -> int:
```

File: app/services/moderation.py [hunk 8]
```diff
@@ -125,17 +125,17 @@ def get_duration(text: str):
...
     chat_member = await bot.get_chat_member(chat_id=chat.chat_id, user_id=user.tg_id)
-    if chat_member.status == ChatMemberStatus.RESTRICTED:
+    if chat_member.status == "restricted":
         assert isinstance(chat_member, ChatMemberRestricted)
         return not chat_member.can_send_messages
-    return chat_member.status in (ChatMemberStatus.BANNED, ChatMemberStatus.KICKED)
+    return chat_member.status in ("banned", "kicked")
 
 
 async def auto_restrict(target: User, chat: Chat, bot: Bot, using_db=None) -> typing.Tuple[int, ModeratorEvent]:
```
Comment (Eclips4) [inline app/services/moderation.py:134]: I don't know, what's actually mean arg `using_db`, but perhaps something like that:
```suggestion
async def auto_restrict(target: User, chat: Chat, bot: Bot, using_db: typing.Optional[typing.Literal[True]] = None) -> typing.Tuple[int, ModeratorEvent]:
```

File: app/models/db/user.py (hunk 3 above)
Comment (bomzheg) [reply app/models/db/user.py:70]: it's ok. in near future i will drop support of 3.10 python too. It is application and not library, we need to support only one version of python

File: app/services/moderation.py (hunk 8 above)
Comment (bomzheg) [reply app/services/moderation.py:134]: using_db is not boolean. it something like connection for make more than one query in trasaction. fucking tortoise=(
probably we should use some other name, but it name used in orm.. I don't know=(

File: app/services/find_target_user.py (hunk 4 above)
Comment (bomzheg) [reply app/services/find_target_user.py:19]: I was fix this file, but all codebase is too large. I think this changes too much increase git diff for this PR

File: app/services/find_target_user.py (hunk 5 above)
Comment (bomzheg) [reply app/services/find_target_user.py:44]: fixed, thanks

File: app/services/find_target_user.py [hunk 9]
```diff
@@ -56,7 +57,7 @@ def has_target_user(target_user: types.User, author_user: types.User, can_be_sam
     return True
 
 
-def is_one_user(user_1: types.User, user_2: types.User):
+def is_one_user(user_1: dto.TargetUser, user_2: dto.TargetUser):
```
[the remark this answers is not in the document]
Comment (bomzheg) [reply app/services/find_target_user.py:60]: fixed, thanks

File: app/services/moderation.py (hunk 6 above)
Comment (bomzheg) [reply app/services/moderation.py:126]: fixed, thanks

File: app/handlers/errors.py (hunk 1 above)
Comment (bomzheg) [reply app/handlers/errors.py:27]: fixed, thanks

File: app/middlewares/db_middleware.py (hunk 2 above)
Comment (bomzheg) [reply app/middlewares/db_middleware.py:37]: remove this cast, probably it will be ok

File: app/services/moderation.py (hunk 6 above)
Comment (bomzheg) [reply app/services/moderation.py:126]: fixed, thanks

File: app/services/moderation.py (hunk 7 above)
Comment (bomzheg) [reply app/services/moderation.py:164]: fixed, thanks

**Review by Eclips4 — APPROVED** [review]

**Pull request merged**